src/exporters/staking.ts

```diff
@@ -0,0 +1,233 @@
+import * as PromClient from "prom-client"
+import {  decimals, getFinalizedApi } from '../index';
+import { logger} from '../logger';
+import { Exporter } from './IExporter';
+import { ApiPromise } from "@polkadot/api";
+import { Header } from "@polkadot/types/interfaces";
+import BN from "bn.js";
+import { ApiDecoration } from "@polkadot/api/types";
+import { AccountId, Balance, } from "@polkadot/types/interfaces/runtime"
+import { PalletBagsListListNode } from "@polkadot/types/lookup"
+
+class StakingExporter implements Exporter {
+	palletIdentifier: any;
+
+	nominatorCountMetric: PromClient.Gauge<"type" | "chain">;
+	validatorCountMetric: PromClient.Gauge<"type" | "chain">;
+	stakeMetric: PromClient.Gauge<"type" | "chain">;
+	ledgerMetric: PromClient.Gauge<"type" | "chain">;
+
+	voterListBags: PromClient.Gauge<"type" | "chain">;
+	voterListNodesPerBag: PromClient.Gauge<"bag" | "chain">;
+	voterListNodes: PromClient.Gauge<"type" | "chain">;
+
+	constructor(registry: PromClient.Registry) {
+		this.palletIdentifier = "staking";
+
+	// staking
+	this.nominatorCountMetric = new PromClient.Gauge({
+		name: "runtime_nominator_count",
+		help: "Total number of nominators in staking system",
+		labelNames: ["type", "chain"]
+	})
+
+	this.validatorCountMetric = new PromClient.Gauge({
+		name: "runtime_validator_count",
+		help: "Total number of validators in staking system",
+		labelNames: ["type", "chain"]
+	})
+
+	this.stakeMetric = new PromClient.Gauge({
+		name: "runtime_stake",
+		help: "Total amount of staked tokens",
+		labelNames: ["type", "chain"]
+	})
+
+	this.ledgerMetric = new PromClient.Gauge({
+		name: "runtime_staking_ledger",
+		help: "the entire staking ledger data",
+		labelNames: ["type", "chain"]
+	})
+
+	// bags-list pallet.
+	this.voterListBags = new PromClient.Gauge({
```

Comment:
We can move the voter-list stuff out of staking for the sake of simplicity.

Reply:
the voter-list part is composed of 2 pallet calls, staking and bagsList. These calls are nested in the same votersList function in the staking exporter.
That being said, the bagsList pallet can exist only if there is a staking pallet, so I suggest that we let it as is. We keep the concept of having one exporter per pallet, but we also tolerate pallets that are intrinsically compatible in some cases like staking.

Ideally, we could add support for multiple pallets into one exporter.